metadata-db-app/src/cmr/metadata_db/data/oracle/concepts.clj

```diff
@@ -575,25 +575,23 @@
          ;; older than the tombstone - up to 'limit' concepts.
          sql (if small
                (format "select t1.concept_id, t1.revision_id from %s t1 inner join
-                         (select * from
                          (select concept_id, revision_id from %s
-                         where provider_id = '%s' and DELETED = 1 and REVISION_DATE < ?)
-                         where rownum < %d) t2
+                         where provider_id = '%s' and DELETED = 1 and REVISION_DATE < ?
+                         FETCH FIRST %d ROWS ONLY) t2
                          on t1.concept_id = t2.concept_id and t1.REVISION_ID <= t2.revision_id"
                        table table provider-id limit)
                (format "select t1.concept_id, t1.revision_id from %s t1 inner join
-                         (select * from
                          (select concept_id, revision_id from %s
-                         where DELETED = 1 and REVISION_DATE < ?)
-                         where rownum < %d) t2
+                         where DELETED = 1 and REVISION_DATE < ?
+                         FETCH FIRST %d ROWS ONLY) t2
                          on t1.concept_id = t2.concept_id and t1.REVISION_ID <= t2.revision_id"
                        table table limit))
          stmt [sql (cr/to-sql-time tombstone-cut-off-date)]
          result (su/query conn stmt)]
      ;; create tuples of concept-id/revision-id to remove
-     (map (fn [{:keys [concept_id revision_id]}]
-            [concept_id revision_id])
-          result))))
+     (doall (map (fn [{:keys [concept_id revision_id]}]
```

Comment:
what side effects are you forcing here?

Reply:
I want to make sure the query gets executed immediately, rather than potentially get delayed because the system is under stress. Remember I was trying to figure out an intermittent outage issue.